src/Orleans/Runtime/GrainReference.cs

```diff
@@ -701,9 +712,24 @@ protected GrainReference(SerializationInfo info, StreamingContext context)
             var genericArg = info.GetString("GenericArguments");
             if (String.IsNullOrEmpty(genericArg))
                 genericArg = null;
-            genericArguments = genericArg;
+            genericArgs = genericArg;
         }
 
         #endregion
+
+        //used for testing
+        string IGrainReferenceInternal.GenericArguments {
+            get {
+                return this.GenericArguments;
+            }
+        }
     }
+
+
+
+    internal interface IGrainReferenceInternal
```

Comment:
I think if this new interface is only used for testing, its better not to pollute the code. you can add internal property `GenericArgumentsForTesting`, but  no need in a new interface.